packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -564,18 +563,19 @@ - (void)captureOutput:(AVCaptureOutput *)output
       _lastAudioSampleTime = currentSampleTime;
 
       if (_audioTimeOffset.value != 0) {
-        CFRelease(sampleBuffer);
-        sampleBuffer = [self adjustTime:sampleBuffer by:_audioTimeOffset];
+        CMSampleBufferRef adjustedSampleBuffer =
+            [self copySampleBufferWithAdjustedTime:sampleBuffer by:_audioTimeOffset];
+        [self newAudioSample:adjustedSampleBuffer];
+        CFRelease(adjustedSampleBuffer);
+      } else {
+        [self newAudioSample:sampleBuffer];
       }
-
-      [self newAudioSample:sampleBuffer];
     }
-
-    CFRelease(sampleBuffer);
   }
 }
 
-- (CMSampleBufferRef)adjustTime:(CMSampleBufferRef)sample by:(CMTime)offset CF_RETURNS_RETAINED {
+- (CMSampleBufferRef)copySampleBufferWithAdjustedTime:(CMSampleBufferRef)sample
+                                                   by:(CMTime)offset CF_RETURNS_RETAINED {
```

Comment:
Is `CF_RETURNS_RETAINED` needed with the `copy` rename?  I thought that was only needed if it was a `copy` but the name didn't include it.

Reply:
you are right. let me remove it. just wanted to put `copy` in the name so +1 is clear from caller site